### PR TITLE
Changing password and adding profile picture

### PR DESCRIPTION
**Pull request by skim803**
User can change password and add profile picture in ProfileActivity.

User now has an ID and profile picture. (See User.java)

### CODE REVIEW

File: app/src/main/java/twotom/bookhub/ProfileActivity.java [hunk 1]
```diff
@@ -38,12 +64,12 @@ protected void onCreate(Bundle savedInstanceState) {
...
 
-        UserManager userManager = new UserManager();
-        User currentUser = userManager.getCurrentUser();
-
-        usernameView.setText(currentUser.getUsername());
-        emailView.setText(currentUser.getEmail());
-
+        mCurrentUser = new UserManager().getCurrentUser();
+        System.out.println(mCurrentUser.getPicture());
+        if (mCurrentUser.getPicture() != "") {
```
Comment (Kijjakarn) [inline app/src/main/java/twotom/bookhub/ProfileActivity.java]: Don't use identify comparison on Strings. Use isEmpty() method instead.

File: app/src/main/java/twotom/bookhub/ProfileActivity.java [hunk 2]
```diff
@@ -38,12 +64,12 @@ protected void onCreate(Bundle savedInstanceState) {
...
             findViewById(R.id.button_profile_password);
 
-        UserManager userManager = new UserManager();
-        User currentUser = userManager.getCurrentUser();
-
-        usernameView.setText(currentUser.getUsername());
-        emailView.setText(currentUser.getEmail());
-
+        mCurrentUser = new UserManager().getCurrentUser();
+        System.out.println(mCurrentUser.getPicture());
```
Comment (Kijjakarn) [inline app/src/main/java/twotom/bookhub/ProfileActivity.java]: Don't print this out in the console.

File: app/src/main/java/twotom/bookhub/User.java [hunk 3]
```diff
@@ -9,16 +9,19 @@
  */
 public class User implements Serializable
 {
+    private String _id;
```
Comment (Kijjakarn) [inline app/src/main/java/twotom/bookhub/User.java]: Don't store user ID in the app. It is specific to the server and the database.

File: app/src/main/java/twotom/bookhub/ProfileActivity.java [hunk 4]
```diff
@@ -105,7 +131,85 @@ public void onChangePassword(View view) {
...
+            try {
+                //TODO: handle error
+                bitmap = MediaStore.Images.Media.getBitmap(this.getContentResolver(), selectedImage);
+            } catch (FileNotFoundException e) {
+                //TODO: handle error
+                e.printStackTrace();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            String picture = BitMapToString(bitmap);
```
Comment (Kijjakarn) [inline app/src/main/java/twotom/bookhub/ProfileActivity.java]: Why does the picture need to be converted back and forth between string and bitmap?

File: app/src/main/java/twotom/bookhub/ProfileActivity.java (hunk 2 above)
Comment (skim803) [reply app/src/main/java/twotom/bookhub/ProfileActivity.java]: Fixed. My bad :(

File: app/src/main/java/twotom/bookhub/ProfileActivity.java (hunk 1 above)
Comment (skim803) [reply app/src/main/java/twotom/bookhub/ProfileActivity.java]: Fixed.

**Pull request merged**